mtprotoproxy.py

```diff
@@ -861,13 +864,82 @@ def set_instant_rst(sock):
     return False
 
 
+async def handle_proxy_protocol(reader, writer):
+    PROXY_SIGNATURE = b"\x50\x52\x4F\x58\x59\x20"
+    PROXY_TCP4 = b"TCP4"
+    PROXY_TCP6 = b"TCP6"
+    PROXY_UNKNOWN = b"UNKNOWN"
+
+    PROXY2_SIGNATURE = b"\x0d\x0a\x0d\x0a\x00\x0d\x0a\x51\x55\x49\x54\x0a"
+    PROXY2_AF_UNSPEC = 0x0
+    PROXY2_AF_INET = 0x1
+    PROXY2_AF_INET6 = 0x2
+
+    header = await reader.readexactly(16)
```

Comment:
there we can read 6 bytes first to check first version of protocol. If it is not the first version we can read 10 more bytes. With these precise readings it is possible to make this function return only the peer address.

Reply:
it depends on first issue - will hande_proxy_protocol() fail on non-proxy header data or not. with configurable behavior (PROTOCOL_PROXY_ONLY) it will still have to return read data for gracefull fallback to handshake. in this case, reading 16 bytes is slightly more optimal because reading 6 first will result in additional read (syscall if not bufferent, not familar with asyncio internal mechaincs) for every v2 connection.